src/common/com/intellij/plugins/haxe/lang/lexer/HaxeConditionalCompilationLexerSupport.java

```diff
@@ -283,8 +282,7 @@ public boolean currentContextIsActive() {
   public void processConditional(CharSequence chars, IElementType type) {
     if (PPIF.equals(type)) {
       // Start a new section...
-      Section newSection = new Section(currentContext, type);
-      currentContext = newSection;
+      currentContext = new Section(currentContext, type);
```

Comment:
Why fix stuff like this?  I get that the code is simpler, but you're not even fixing a bug here.

I leave code like this in place because at some point I needed to set a breakpoint and inspect the variable (in this case to see the difference between currentContext and the new Section before the assignment occurred).  I figure that if I needed the breakpoint in the past, I'll probably need it again in the future.  Besides that, there is *zero* performance penalty for having the extra variable -- the JIT compiles it out.

Reply:
There is no difference, but I have no idea why to leave it splitted if there is no need anymore? Anyway I'll revert it, but I don't think that "maybe I would need a breakpoint here" as a good reason to leave code exploded. Why not to explode all declarations and their initialization then?